src/tokenizer.rs

```diff
@@ -35,7 +35,7 @@ pub enum Token {
     /// A keyword (like SELECT) or an optionally quoted SQL identifier
     Word(Word),
     /// An unsigned numeric literal
-    Number(String),
+    Number(String, bool),
```

Comment:
I think this could be better extracted to a named field, e.g. `long: bool`?.
This helps readability